services/workflows-service/src/customer/api-key/api-key.service.intg.test.ts

```diff
@@ -0,0 +1,140 @@
+import { PrismaModule } from '@/prisma/prisma.module';
+import { PrismaService } from '@/prisma/prisma.service';
+import { createCustomerOnly } from '@/test/helpers/create-customer';
+import { createProject } from '@/test/helpers/create-project';
+import { cleanupDatabase, tearDownDatabase } from '@/test/helpers/database-helper';
+import { initiateNestApp } from '@/test/helpers/nest-app-helper';
+import { faker } from '@faker-js/faker';
+import { INestApplication } from '@nestjs/common';
+import { Customer, PrismaClient, Project } from '@prisma/client';
+import { ClsModule } from 'nestjs-cls';
+import { ApiKeyService } from './api-key.service';
+import { ApiKeyRepository } from './api-key.repository';
+import { encryptApiKey, generateApiKey } from './utils';
+
+describe('#ApiKeyService', () => {
+  let app: INestApplication;
+  let apiKeyService: ApiKeyService;
+  let prismaClient: PrismaClient;
+  let customer: Customer;
+  let project: Project;
+
+  beforeAll(cleanupDatabase);
+  afterEach(tearDownDatabase);
+
+  beforeAll(async () => {
+    await cleanupDatabase();
+
+    const servicesProviders = [ApiKeyService, ApiKeyRepository, PrismaService];
+
+    app = await initiateNestApp(app, servicesProviders, [], [PrismaModule, ClsModule]);
+
+    apiKeyService = app.get(ApiKeyService);
+    prismaClient = app.get(PrismaService);
+  });
+
+  beforeEach(async () => {
+    customer = await createCustomerOnly(
+      prismaClient,
+      faker.datatype.uuid(),
+      '',
+      '',
+      'webhook-shared-secret',
+    );
+
+    project = await createProject(prismaClient, customer, faker.datatype.uuid());
+  });
+
+  it('creates same a api key for customer using a salt should throw a unique constraint error', async () => {
+    const apiKey1 = await apiKeyService.createHashedApiKey(customer.id, { apiKey: 'blabla' });
+
+    expect(apiKey1).toBeDefined();
+    expect(apiKey1).toMatchObject({
+      hashedKey: '0de819cc2ab05e54f9c83352fb0130b74e3eea07c53ed563173200be18b8ff6e',
```

Comment:
This test will be flaky as every dev might have a different hashing key. To make you code more testable, consider getting the hashing key as a parameter.